pipeline/runtime/runtestsV2.go

```diff
@@ -0,0 +1,330 @@
+// Copyright 2024 Drone.IO Inc. All rights reserved.
+// Use of this source code is governed by the Polyform License
+// that can be found in the LICENSE file.
+
+package runtime
+
+import (
+	"context"
+	"fmt"
+	"io"
+	"os"
+	"path/filepath"
+	"time"
+
+	"github.com/drone/runner-go/pipeline/runtime"
+	"github.com/harness/lite-engine/api"
+	"github.com/harness/lite-engine/engine"
+	"github.com/harness/lite-engine/internal/filesystem"
+	"github.com/harness/lite-engine/pipeline"
+	"github.com/harness/lite-engine/ti/callgraph"
+	tiCfg "github.com/harness/lite-engine/ti/config"
+	utils "github.com/harness/lite-engine/ti/instrumentation"
+	"github.com/harness/lite-engine/ti/instrumentation/java"
+	"github.com/harness/lite-engine/ti/report"
+	"github.com/harness/ti-client/types"
+	"github.com/sirupsen/logrus"
+)
+
+const (
+	outDir           = "%s/ti/new/callgraph/" // path passed as outDir in the config.ini file
+	javaNewAgentArg  = "-javaagent:%s=%s"
+	javaNewAgentJar  = "java-agent-trampoline-0.0.1-SNAPSHOT.jar"
+	javaNewAgentPath = "/java/new/"
+	javaNewAgentUrl  = "https://raw.githubusercontent.com/ShobhitSingh11/google-api-php-client/4494215f58677113656f80d975d08027439af5a7/java-agent-trampoline-0.0.1-SNAPSHOT.jar" //May be changed later
+	filterDir        = "ti/new/callgraph"
+)
+
+// Ignoring optimization state for now
+func executeRunTestsV2Step(ctx context.Context, engine *engine.Engine, r *api.StartStepRequest, out io.Writer, tiConfig *tiCfg.Cfg) (*runtime.State, map[string]string, map[string]string, []byte, []*api.OutputV2, string, error) {
+	start := time.Now()
+	tmpFilePath := tiConfig.GetDataDir()
+	fs := filesystem.New()
+	log := logrus.New()
+	log.Out = out
+	preCmd, err := getPreCmd(tmpFilePath, fs, log) // Setting up JAVA_TOOL_OPTIONS, configfile
+	if err != nil {
+		return nil, nil, nil, nil, nil, "", fmt.Errorf("failed to set config file or env variable to inject agent")
+	}
+
+	err = downloadJavaAgent(ctx, tmpFilePath, fs, log)
+	if err != nil {
+		return nil, nil, nil, nil, nil, "", fmt.Errorf("failed to download Java agent")
+	}
+
+	commands := fmt.Sprintf("%s\n%s", preCmd, r.RunTestsV2.Command[0])
+	step := toStep(r)
+	step.Command = []string{commands}
+	step.Entrypoint = r.RunTestsV2.Entrypoint
+	setTiEnvVariables(step, tiConfig)
```

Comment:
Can you not do the export of JAVA_TOOL_OPTIONS here instead of returning it from `getPreCmd` and sending explicitly to `step.Command` ?

Reply:
Since for preCommand we needed iniFilePath being coming from javaAgentConfigFile and also same inifile is being used to write to .bazelrc file to support bazel execution. That's why we have enclosed in a method where config file is being created and env variable is being set